### PR TITLE
Log errors from external chown script.

### PR DESCRIPTION
**Pull request by Petraea**
Hi there,
I noticed this TODO here related to helping debug external chown scripts. I've just been debugging this, so this patch helped me to figure out what is going on.

Does this need tests attached?

### CODE REVIEW

File: lib/galaxy/jobs/__init__.py [hunk 1]
```diff
@@ -1755,8 +1755,11 @@ def _change_ownership(self, username, gid):
             cmd.extend([self.working_directory, username, str(gid)])
             log.debug('(%s) Changing ownership of working directory with: %s' % (job.id, ' '.join(cmd)))
             p = subprocess.Popen(cmd, shell=False, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-            # TODO: log stdout/stderr
             stdout, stderr = p.communicate()
+            if p.returncode != 0:
+                logging.error('external script failed.')
+                logging.error('stdout was: %s' %stdout)
+                logging.error('stderr was: %s' %stderr)
```
Comment (nsoranzo) [inline lib/galaxy/jobs/__init__.py]: s/logging/log/ in the 3 lines above.

Also there's a missing space after the `%` operator in the 2 lines above.

**Review by nsoranzo — COMMENTED** [review]
Nice to hear from you @Petraea!

File: lib/galaxy/jobs/__init__.py (hunk 1 above)
Comment (Petraea) [reply lib/galaxy/jobs/__init__.py]: Hi Nick! Took me a few months before I could get our instance free enough to update it.

Good call - made a new commit.

**Comment by nsoranzo** [comment]
@galaxybot test this

**Comment by nsoranzo** [comment]
@Petraea I've re-targeted the PR to the `dev` branch, since this looks more like an enhancement than a bugfix, please scream if you actually need this in 17.09 and we can reconsider.

**Comment by nsoranzo** [comment]
Thanks @Petraea!

**Pull request merged**